app/User.php

```diff
@@ -32,4 +34,33 @@ public function roles()
     {
         return $this->belongsToMany(Role::class);
     }
+
+    public function tokens()
+    {
+        return $this->hasMany(Token::class, 'user_id');
+    }
+
+    public static function getUserIdFromToken(Request $request)
+    {
+        $result = null;
+        if ($request->header('Authorization')) {
+            $token = self::sendTokenToPassport($request);
+            if ($token->getStatusCode() != Response::HTTP_UNAUTHORIZED) {
+                $result = json_decode($token->getBody())->id;
+            }
+        }
+        return $result;
+    }
+
+    public static function sendTokenToPassport(Request $request)
```

Comment:
this one the same

Reply:
I do not know how to get a middleware feature so I can call it from the model.